src/less/dialog/ds6/dialog-base.less

```diff
@@ -0,0 +1,209 @@
+// BLOCK
+//-----------------------------
+
+.dialog {
+    .a11y-require(~'[role="dialog"]');
```

Comment:
@ianmcburnie so this is what I was talking about the other day. This mixin [located here](https://github.com/eBay/skin/pull/113/files#diff-ad2c98e93249cabc93e32425ddb3da67R122) ensures that the provided selector matches on the element and otherwise shows an inline error message like the following:

![screen shot 2018-04-03 at 12 43 20 pm](https://user-images.githubusercontent.com/4985201/38271931-a7ea5402-373c-11e8-97a0-a0332c51325c.png)

Again this would only show up when the provided selector (in this case `role=document`) is missing and is a bit more helpful than the current approach of simply not applying the styles. Also it has the added benefit of not increasing specificity of the primary selectors while ensuring the appropriate accessibility attrs/tags are used.

Reply:
Yep. There are pros and cons to both approaches. I don't think this should be part of this PR. If we are going to change our a11y strategy it should be discussed first separately.

Reply:
One con of this approach is that we have to write additional CSS. And that is not CSS we would want to send to the client. So we would need a 'validator.css` or the like, that only gets included in dev build/environment, but my fear would be that devs sidestep this validator. Also, we have to think about static sites that use CDN.

Reply:
I had actually meant to mention that as well. Ideally this mixin would be removed in prod which we could do using a lasso flag like so:

```
"if": "process.env.NODE_ENV && process.env.NODE_ENV !== 'development'"
```

Also technically we still have additional CSS in prod right now (for the aria attributes) but we could actually completely remove these checks in prod with the above approach.

I'm fine with making this a separate PR, I was just hoping it would be possible to squeeze in since we are releasing a major anyway. I hope to continue this discussion, I don't believe it would be much work to change the existing less to work this way as well if we decided to transition.